test/IDE/print_ast_tc_decls.swift

```diff
@@ -1343,7 +1343,8 @@ public func ParamAttrs5(a : (@escaping () -> ()) -> ()) {
 // PASS_PRINT_AST: public typealias ParamAttrs6 = (@autoclosure () -> ()) -> ()
 public typealias ParamAttrs6 = (@autoclosure () -> ()) -> ()
 
-// PASS_PRINT_AST: public var ParamAttrs7: (@escaping () -> ()) -> ()
+// PASS_PRINT_AST_TYPE: public var ParamAttrs7: (_ f: @escaping () -> ()) -> ()
+// PASS_PRINT_AST_TYPEREPR: public var ParamAttrs7: (@escaping () -> ()) -> ()
```

Comment:
Could you fix the TyprRepr printing as well?

Reply:
> Could you fix the TyprRepr printing as well?
I think the current behaviour is intended. We are printing the following decl

```swift
public var ParamAttrs7: (@escaping () -> ()) -> () = { f in f() }
```

We only get the internal parameter name f through type inference from the closure after =. IIUC this affects the Type but not the TypeRepr which matches what is written in source code.
Printing the AST of

```swift
public var ParamAttrs7: (_ f: @escaping () -> ()) -> () = { f in f() }
```
with `-prefer-type-repr=true` does print the internal name.

I added another test to test the second case.